src/Csv.php

```diff
@@ -444,14 +444,28 @@ public function output($filename = null, $data = array(), $fields = array(), $de
             $mime = $delimiter === "\t" ?
                 'text/tab-separated-values' :
                 'application/csv';
+
+            // Buffer output:
+            ob_start();
```

Comment:
I suppose we shouldn't nest `ob_start` calls. Also, they need to come in pairs with `ob_end_*` calls. We need output buffering only if gzip is supported.